### PR TITLE
Eburgess gradio lab

### PR DESCRIPTION
**Pull request by EmanuelB25**
adding Gradio chatbot

### CODE REVIEW

File: ai-ml/gemini-chatbot-app/gradio/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+ARG GRADIO_SERVER_PORT=8080
+
+COPY requirements.txt ./
+
+COPY gradio_app.py ./
+
+RUN pip3 install -r requirements.txt
+
+# Dirty fix for https://github.com/gradio-app/gradio/pull/7707 regression
```
Comment (rsamborski) [inline ai-ml/gemini-chatbot-app/gradio/Dockerfile]:
```suggestion
# Dirty fix, won't be necessary when https://github.com/gradio-app/gradio/pull/8962 gets into Gradio release 
```

File: ai-ml/gemini-chatbot-app/gradio/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Purpose: To deploy the App to Cloud Run.
+
+# Google Cloud Project
+PROJECT=emanuel-ai
```
Comment (rsamborski) [inline ai-ml/gemini-chatbot-app/gradio/deploy.sh]:
```suggestion
PROJECT=<YOUR-PROJECT-ID>
```

File: ai-ml/gemini-chatbot-app/gradio/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Purpose: To deploy the App to Cloud Run.
+
+# Google Cloud Project
```
Comment (rsamborski) [inline ai-ml/gemini-chatbot-app/gradio/deploy.sh]:
```suggestion
# @TODO Replace <YOUR-PROJECT-ID> with your Google Cloud Project ID
```

**Review by rsamborski — CHANGES_REQUESTED** [review]
Thanks for the PR. I suggest to move it to a different folder, as currently you're putting it together with Sampath's streamlit bot. Potential location could be: `ai-ml/gemini-chatbot-app-gradio`, `ai-ml/gemini-chatbot-gradio` or something similar.

Couple other suggestions inline.

File: ai-ml/gemini-chatbot-app/gradio/deploy.sh (hunk 3 above)
Comment (CadillacBurgess1) [reply ai-ml/gemini-chatbot-app/gradio/deploy.sh]: updated. Thanks I missed this.

File: ai-ml/gemini-chatbot-app/gradio/deploy.sh (hunk 2 above)
Comment (CadillacBurgess1) [reply ai-ml/gemini-chatbot-app/gradio/deploy.sh]: updated. Thanks I missed this.

File: ai-ml/gemini-chatbot-app/gradio/Dockerfile (hunk 1 above)
Comment (CadillacBurgess1) [reply ai-ml/gemini-chatbot-app/gradio/Dockerfile]: updated. Thanks I missed this.

**Comment by EmanuelB25** [comment]
I did a push to fix issues. Please review.

**Comment by thagomizer** [comment]
Can you please add Apache license headers and then I'll merge.

**Comment by rsamborski** [comment]
> Can you please add Apache license headers and then I'll merge.

@thagomizer Good catch. 
@EmanuelB25 Please also remember to remove duplicate files in folder `ai-ml/gemini-chatbot-app/gradio`, you should only preserve files in `ai-ml/gemini-chatbot-app-gradio`.

**Comment by EmanuelB25** [comment]
Updated. @rsamborski

**Review by rsamborski — APPROVED** [review]

**Pull request merged**